baybe/constraints/continuous.py

```diff
@@ -115,25 +130,61 @@ def to_botorch(
         Args:
             parameters: The parameter objects of the continuous space.
             idx_offset: Offset to the provided parameter indices.
+            batch_size: The batch size used for the recommendation. Necessary
+                for interpoint constraints as indices need to be adjusted.
+                Ignored by all other constraints.
 
         Returns:
             The tuple required by botorch.
+
+        Raises:
+            RuntimeError: When the constraint is an interpoint constraint but
+                ``batch_size`` is ``None``.
         """
         import torch
 
         from baybe.utils.torch import DTypeFloatTorch
 
+        # NOTE: The interpoint constraint case requires indices to be a 2-d tensor.
+        # This requires some adjustments to the indices.
+        # See https://botorch.readthedocs.io/en/latest/optim.html, in particular the
+        # docstring of ``optimize_acqf`` for details.
+
         param_names = [p.name for p in parameters]
-        param_indices = [
-            param_names.index(p) + idx_offset
-            for p in self.parameters
-            if p in param_names
-        ]
+        coefficients: list[float]
+        torch_indices: Tensor
+        if not self.is_interpoint:
+            param_indices = [
+                param_names.index(p) + idx_offset
+                for p in self.parameters
+                if p in param_names
+            ]
+            coefficients = self.coefficients
+            torch_indices = torch.tensor(param_indices)
+        elif batch_size is None:
```

Comment:
Instead of mixing logic and validation, let's do the cleaner and easier-to-read guard-clause approach, where you validate all input before you turn toward computation